src/libraries/Microsoft.XmlSerializer.Generator/tests/SGenTests.cs

```diff
@@ -15,13 +15,19 @@ public static class SgenTests
         [Fact]
         public static void SgenCommandTest()
         {
-            string codefile = "Microsoft.XmlSerializer.Generator.Tests.XmlSerializers.cs";
+            const string CodeFile = "Microsoft.XmlSerializer.Generator.Tests.XmlSerializers.cs";
+            long lkgSize = new System.IO.FileInfo(CodeFile).Length;
+
             var type = Type.GetType("Microsoft.XmlSerializer.Generator.Sgen, dotnet-Microsoft.XmlSerializer.Generator");
             MethodInfo md = type.GetMethod("Main", BindingFlags.Static | BindingFlags.Public);
             string[] args = new string[] { "Microsoft.XmlSerializer.Generator.Tests.dll", "--force", "--quiet" };
             int n = (int)md.Invoke(null, new object[] { args });
+
             Assert.Equal(0, n);
-            Assert.True(File.Exists(codefile), string.Format("Fail to generate {0}.", codefile));
+            Assert.True(File.Exists(CodeFile), string.Format("Fail to generate {0}.", CodeFile));
+            // Compare the generated CodeFiles from the LKG with the live built shared framework one.
+            // Not comparing byte per byte as the generated output isn't deterministic.
```

Comment:
Fields are written in a random order (presumably because reflection behaves differently based on the references loaded. I observed the same behavior when I implemented the binary serialization tests, in the past).

Reply:
Huh, that's odd.  I could have sworn that reflection was deterministic.  I know metadata order is deterministic.  I remember some bugs in the past where different machines sorted files differently so we passed them in a different order to the compiler and we saw different metadata ordering based on platform/Culture.  Anyhow, this seems ok.